molgenis/molgenis-emx2
# fix: upload in inherited tables should throw duplicate key if same key is used in different tables

This fixes that issue that if you have subclasses, and then identifiers of rows in the subclass tables are the same that it will throw an error (instead of silently overwriting). See example of Brenda below, about SourceVariables and RepeatedSourceVariables that have the same name and then one hides the other.

fixes #824. 
fixes #594. 
possibly also fixes #1002 
Also the use case in catalogue staging, where RepeatedSourceVariables had overwritten SourceVariables
Note: we don't use this ontology structure but bug also emerges in catalogue model with inherited tables.

This PR contains:
- trigger to check and prevent updates on another mg_tableclass than inserted
- fix of test data set that did not conform to this

To test:
- create a main table 'pets' with primary key 'name'
- create an extended table 'cats' extends 'pets'
- using csv/excel: upload a 'pooky' into pets and also into 'cats'

behavior before the fix: 
- you will only see a record in 'cats', the 'pets' is empty

behavior after fix: 
- you will receive duplicate key exception on 'cats' that pooky already exists in 'pets'
